### ISSUE TITLE
[Accessibility]: Branch/tag/commit: Non-interactive element is focusable in header

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35475]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34417

### Problem description

<img width="620" alt="image" src="https://user-images.githubusercontent.com/9516420/168597088-5ff5c450-c733-488d-97b1-cdc8dd10e87b.png">

Note the blue focus line. This element is not interactive and should not be accessed via the tab key.

### Expected behavior

Element is ignored when tabbing through the page.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by gitstart-sourcegraph** [comment]
Hi @umpox 
To resolve this issue we thought of 2 options.
1. Support a new `tabIndex` prop on `TabPanel`, which defaults to `-1` to ensure that the container is not focusable (which is the reason for the issue stated above.). The limitation of this approach is that screen reader won't read `... tab panel` when users navigate from tablist to tabpanel. 
2. we could update the UI of panel in focus-visible state instead to remove box-shadow, the limitation here is that, the container would be focusable (Although not visible)

WDYT?

**Comment by umpox** [comment]
@gitstart-sourcegraph Thanks, I see this is quite more detailed than originally thought. Looks like it's good practice to make tabs focusable when the content is non-interactive copy, we can make an exception in this case.

> Support a new tabIndex prop on TabPanel, which defaults to -1 to ensure that the container is not focusable (which is the reason for the issue stated above.). The limitation of this approach is that screen reader won't read ... tab panel when users navigate from tablist to tabpanel.

`TabPanel` should already support overriding `tabIndex`, let's set `tabIndex={-1}` on `TabPanel` in `RevisionsPopover`

**Comment by umpox** [comment]
Closed through https://github.com/sourcegraph/sourcegraph/pull/38203
